[ADD] l10n_uy: new module

zaoral opened the pull request:
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:




---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/133891).

zaoral commented:
HI @xmo-odoo
Thank you for the review

I have applied the suggested changes.
Please review and let me know if there is anything else pending

Best Regards

zaoral commented on addons/l10n_uy/models/account_move.py:
```diff
@@ -0,0 +1,87 @@
+            if len(vat_taxes) != 0:
+                raise ValidationError(
+                    _('Should not be any VAT tax in the export cfe line "%s" (Id Invoice: %s)' % (
+                        inv_line.product_id.name, inv_line.move_id.id)))
```
@xmo-odoo 
Here I have applied the suggested changes, taking into account what is explained [here](https://www.odoo.com/documentation/16.0/developer/howtos/translations.html?highlight=translation)

But not sure I have applied correctly, Please let me know if it is ok or if I need to change something else

Best Regards

fvz-odoo commented:
Hello @jco-odoo 

This is the PR for the base module in Uruguay. Could you please take a look? The error we reported you is still there but as you told us, it is a bug in master. Once it is fixed, I will go ahead and test again just in case.

I tested the following:
- Chart of Accounts: we cleaned it, mapped it and kept the most important accounts
- Contacts: added the type of identification field and validation of the identification format (also changed the message to be more consistent)
- Journals: left out the type of journal field, we will need it for electronic invoicing in the future
- Taxes: we kept what we had and mapped them with the correct accounts
- Document Types: changed the prefix, added internal type and cleaned them overall
- Translations: Checked all of the above in English and in Spanish

- Double-checked the code to make sure it makes sense with what we do in the localizations and cleaned all fields that are not required for this base module.

Let me know if anything

Fernando

@zaoral @ren-odoo @pablohmontenegro

jco-odoo commented on addons/l10n_uy/tests/test_manual.py:
```diff
@@ -0,0 +1,90 @@
+
+
+@tagged('post_install_l10n', '-at_install', 'post_install')
+class TestManual(AccountTestInvoicingCommon):
```
It does not seem that useful to have tests just for 2 validations...

zaoral replied:
Yes indeed, there is not a lot to test at this point. We added the unit test because it was requested by Odoo team. We can remove it if you like @jco-odoo  please let me know

jco-odoo replied:
Please remove.

jco-odoo reviewed (commented):
I went quickly through it, so we can quickly progress on this for the freeze.  (we can still do some adaptations after, so if we have too many discussions, we can just leave it out for now)

ren-odoo commented on addons/base_vat/models/res_partner.py:
```diff
@@ -251,7 +252,7 @@ def _build_vat_error_message(self, country_code, wrong_vat, record_label):
             vat_label = company.country_id.vat_label
 
-        expected_format = _ref_vat.get(country_code, "'CC##' (CC=Country Code, ##=VAT Number)")
+        expected_format = _ref_vat.get(country_code, "'##' (##=VAT Number)")
```
Ok, I'll create an internal task to evaluate the improvements to be done on base_vat.

zaoral replied:
Ok, We will remove the change of this PR, please let us know if you likw us to create a new PR with only this change.
Best Regards

@ren-odoo  @jco-odoo

ren-odoo commented on addons/l10n_uy/data/res_partner_data.xml:
```diff
@@ -0,0 +1,17 @@
+<odoo>
+
+    <record id="partner_dgi" model="res.partner">
+        <field name="name">DGI</field>
```
Is the Fiscal office on UY:  Direccion General Impositiva: https://www.dgi.gub.uy/wdgi/page?2,principal,DireccionGeneralImpositiva,O,es,0,

zaoral replied:
I added a comment to adding the info given by  @ren-odoo so this way will clear for everyone that read the files

zaoral commented on addons/l10n_uy/data/res_currency_rate_data.xml:
```diff
@@ -0,0 +1,16 @@
+<?xml version='1.0' encoding='utf-8'?>
+<odoo noupdate="1">
+
+    <record forcecreate="0" id="rateUYU" model="res.currency.rate">
```
Yes is for demo, Also, this way we also trying to make the UYU currency the default principal one. do you want us to remove both rates? and only change the symbol and rounding  of UYO in the base module?

zaoral commented on addons/l10n_uy/data/res_currency_data.xml:
```diff
@@ -0,0 +1,12 @@
+    <!-- It is an official Uruguayan unit of value, which is readjusted according
+    to inflation, measured by the Consumer Price Index (CPI). -->
+    <record id="base.UYI" model="res.currency">
+        <field name="symbol">UYI</field>
+        <field name="rounding">0.0001</field>
```
We will move this to the base module.
Here we only leave the active True in order to activate the currency when the loc is installed.
Please let us know if there is a better way to do it
@jco-odoo

zaoral commented on addons/l10n_uy/__manifest__.py:
```diff
@@ -5,23 +5,66 @@
+
+Follow the steps below to configure in production after installing the module:
+
+1. Company Configuration. Go to the Company menu and create or configure the Uruguayan Company. set the RUT number (vat)
```
indeed. would you like that we remove this part? @jco-odoo

jco-odoo replied:
Yes (the part that is generic knowledge)

zaoral replied:
@jco-odoo 
Ok, Already removed.
I mark this comment as resolved

zaoral commented on addons/l10n_uy/models/account_journal.py:
```diff
@@ -0,0 +1,21 @@
+
+    _inherit = "account.journal"
+
+    def _l10n_uy_get_journal_codes(self):
```
We did similar to what we already have in AR https://github.com/odoo/odoo/blob/17d7b5199f06aa606f3cbdb8c838fcad7047d9c9/addons/l10n_ar/models/account_journal.py#L89 Do you prefer that we move this method to account.move?  @jco-odoo 

Do you prefer

zaoral commented on odoo/addons/base/data/res.country.state.csv:
```diff
@@ -1666,6 +1666,25 @@ state_sa_90,sa,"Yanbu commercial city","YNB"
+state_uy_16,base.uy,San José,UY-SJ
+state_uy_17,base.uy,Soriano,UY-SO
+state_uy_18,base.uy,Tacuarembó,UY-TA
+state_uy_19,base.uy,Treinta y Tres,UY-TT
```
Hi Jos

In EDI documents. e-Factura and some of the e-Tickets documents, are required to have the direction of the partner, including having the country state. We are not adding an EDI module. but we are leaving the base so the user can generate EDI in other software and then save it in Odoo.

Having preload all the country states is just a step to make it easier so the user should not need to define them by himself with the risk of creating duplicated values, etc.

Not sure if you think this is a valid base to have them, Please  @fvz-odoo @ren-odoo give us you feedback.
At last, if you want us to remove it let us know and we will remove them @jco-odoo

ren-odoo replied:
Hi @jco-odoo, I consider @zaoral arguments are valid enough to keep the data in the base module. 

cc. @fvz-odoo

zaoral commented on addons/l10n_uy/models/account_move.py:
```diff
@@ -0,0 +1,104 @@
+            ('module', '=', 'account'),
+            ('model', '=', 'account.tax.group'),
+            '|', '|',
+            ('name', 'ilike', 'tax_group_iva_10'),
```
Hi @jco-odoo 

At the current moment for this module we needed in order to identify which invoice lines have UY vat taxes. to ensure that we only have one Uruguayan vat tax per line. Also we are checking that not vat taxes are added to Export Invoices documents.

In future  l10n_uy_reports and l10n_uy_edi Odoo modules , we use them to identify and match the taxes with the columns of the Uruguayan VAT book. to generate some TXT and for electronic invoicing to properly enconde the vat taxes on the xml.

We talk about this with @fvz-odoo  @ren-odoo on meeting, I think that we will recommed the user to the the current tax groups not to create new ones

ren-odoo replied:
Hi @zaoral the scalidraw that @fvz-odoo shared https://app.excalidraw.com/l/4zcUIVVimSO/4yL8A96tUCJ  is actually to cover this use case. By adding the selection field l10n_uy_type, it will allow us to have an easy way to identify the  VAT taxes or any other Tax type and take this as a parameter for future reports or EDI data. 

cc. @jco-odoo

zaoral replied:
Hi @ren-odoo
Yes indeed, I have added the suggested field.
My last comment was meant to be to try answer Jos "(I still need to see why we need it however) What if a user creates a tax group by himself?" 

Now that we have the field we can mark this comment as resolve?
please let me know best regards

zaoral commented on addons/l10n_uy/models/account_move.py:
```diff
@@ -0,0 +1,104 @@
+        """ When using documents on vendor bills the document_number is set manually by the number given from the vendor
+        so the odoo sequence is not used. In this case we allow to delete vendor bills with document_number/name """
+        self.filtered(lambda x: x.move_type in x.get_purchase_types() and x.state in ('draft', 'cancel') and
+                      x.l10n_latam_use_documents).write({'name': '/'})
```
yes indeed, actually we have something similar to this in l10n_ar.
Hi  @ren-odoo  @fvz-odoo @vbe-odoo please could you check if this behavior is the same for all the latam localizations that are using l10n_latam_invoice_document module, if it is let me know and we can add it as a generic approach on a new pr

vbe-odoo replied:
@zaoral what I understood is that here we are allowing to delete an invoice after being posted and that a document_number was manually typed in?
Or is it something different? Maybe I am missing something.
Thanks!

jco-odoo replied:
Please remove here for now.  I can not find it back in l10n_ar either.

zaoral replied:
Done. I remove it

zaoral commented on addons/l10n_uy/models/account_journal.py:
```diff
@@ -0,0 +1,21 @@
+            ('country_id', '=', self.env.ref('base.uy').id)])
+        available_types = doc_types.mapped('code')
+
+        return available_types
```
it is already in https://github.com/odoo/odoo/blob/d8a043cc026c7043eb7de0be93a226dcdf8bcf2e/addons/l10n_uy/models/res_company.py#L9

zaoral commented on addons/l10n_uy/models/account_move.py:
```diff
@@ -0,0 +1,104 @@
+            param['l10n_latam_document_type_id'] = self.l10n_latam_document_type_id.id or 0
+        return where_string, param
+
+    def _is_manual_document_number(self):
```
You are right, it is already handle there, but  for some reason is not working. I will check what is happening

zaoral commented on addons/l10n_uy/models/l10n_latam_document_type.py:
```diff
@@ -0,0 +1,30 @@
+
+    _inherit = 'l10n_latam.document.type'
+
+    internal_type = fields.Selection(selection_add=[('stock_picking', 'Delivery Guide')])
```
We can add the selection option directly to l10n_latam_invoice_document module. do you think this it could be a problem? do you prefer that we leave it here?  @jco-odoo

zaoral replied:
Ok. Done. I will create the PR for latam later

zaoral commented on addons/l10n_uy/models/template_uy.py:
```diff
@@ -24,7 +27,6 @@ def _get_uy_res_company(self):
                 'bank_account_code_prefix': '1111',
                 'cash_account_code_prefix': '1112',
                 'transfer_account_code_prefix': '11120',
-                'account_default_pos_receivable_account_id': 'uy_code_11307',
```
we remove it because the account does not exist aymore.
If you think we need to define a default one please @MarianaLuzTorres @fvz-odoo  @ren-odoo could you help us with this?
Best Regards

ren-odoo replied:
@zaoral as agreed with @marianaluzTorres let's add the account back :)

thank you.

zaoral replied:
Thank you all.
I have re-added the account.
Best Regards
@ren-odooh @MarianaLuzTorres @jco-odoo

zaoral commented on addons/l10n_uy/models/template_uy.py:
```diff
@@ -33,3 +35,29 @@ def _get_uy_res_company(self):
+                "refund_sequence": False,
+            },
+            'purchase': {
+                "name": _("Purchases"),
```
changed to the suggestion @jco-odoo .
cc @fvz-odoo @ren-odoo

zaoral commented on addons/l10n_uy/data/res_partner_data.xml:
```diff
@@ -0,0 +1,17 @@
+
+    <record model='res.partner' id='partner_cfu'>
+        <field name='name'>Consumidor Final Anónimo</field>
+        <field name='l10n_latam_identification_type_id' ref='it_dni'/>
```
At the current module status is not required at all.
Not sure if it is required to be defined for EDI. I thnik we are requiring in some place the partner should always have a identification type define of type UY

ren-odoo replied:
@jco-odoo if there is no technical restriction on having the identification type,  I suggest we keep it, as it can save some time when testing worflows on EDI where this field is requiered, and doesn't harm to have set in this base module.

zaoral replied:
Ok, for the moment we leave the identification type.

zaoral replied:
if you have any comment @jco-odoo please let us know

zaoral commented on addons/l10n_uy/models/account_move.py:
```diff
@@ -0,0 +1,104 @@
+        """ Check that the UY invoices are vaid
+        * the partner of the invoice should have a valid identification type
+        * each invoice line should have one and only one vat tax
+        * export invoices should not have any vat tax.
```
We can use VAT 0% (actually VAT Exempt)
From the user and functional point of view will be the same.

But for the technical side, If we want to use Exempt tax on Export Invoices then we need also to add a validation to ensure that they are using Exempt and not another VAT tax. And, If we want it to make it easier for the user maybe we could add some fiscal positions but not sure if we want to add more complexity. @ren-odoo @fvz-odoo @MarianaLuzTorres

More from the technical point of what we have in EDI, Exempt tax for regular invoices we decode them to send some info to DGI taking into account the tax type. For Expo invoices, we used a different logic because we needed to encode the values differently.

ren-odoo replied:
@jco-odoo considering your comments we're plannig on adding the fiscal position approach we did for Peru https://github.com/odoo/odoo/blob/16.0/addons/l10n_pe/data/fiscal_position_data.xml, does it make sense for  you? 

@zaoral @fvz-odoo

zaoral replied:
Hi @jco-odoo @ren-odoo 
I have added the fiscal positions as in PE.
Also, I removed the check for exportation invoices

Please could you review @ren-odoo 
Best Regards

zaoral commented on addons/l10n_uy/models/account_move.py:
```diff
@@ -0,0 +1,104 @@
+
+        uruguayan_vat_taxes = self._get_uruguayan_vat_taxes()
+
+        # Check that we do not send any tax in export invoices
```
samo of about

zaoral replied:
We removed the check for export invoices

zaoral commented on addons/l10n_uy/models/template_uy.py:
```diff
@@ -33,3 +35,29 @@ def _get_uy_res_company(self):
+        """ Set companies country by default after install the chart of account,
+        also set the rut as the company identification type because this one is the uruguayan vat """
+        res = super()._load(template_code, company, install_demo)
+        if template_code == 'uy':
```
We do it for AR loc but not sure is there is a generic code that does it?

https://github.com/odoo/odoo/blob/8e6a3d7269f81191ec8180e6d5ee8d403238f9a2/addons/l10n_ar/models/account_chart_template.py#L21-L45

1. I think that the identification type of the partner-company should be set to default marked as is_vat. There is a generic code that does it? I notice this is made for creating new companies on the l10n_latam_base module but not to existing companies when installing CoA. 

2. related to setting the country on the company. Also think is a good thing to do. You think this could be a problem?

@ren-odoo @fvz-odoo please give us your feedback

ren-odoo replied:
@jco-odoo in the Locs where we don't have correct  the identification type we need to manually update the value when doing the testing, so if we can set as default in the demo is better from the Funtional point of view. 

cc. @fvz-odoo @zaoral

jco-odoo replied:
We might do this in generic as well.  There is a default, which is the generic VAT but we could make it a compute.  For AR,there are multiple CoA, which makes it more logical.  For sure, we should not set the country.  We could keep the identification type, but VAT vs RUT, it should not have much difference in behaviour.

zaoral replied:
@jco-odoo 
Ok. I will remov the country and leave the indentitication type by the moment

zaoral commented on addons/l10n_uy/views/l10n_latam_document_type_views.xml:
```diff
@@ -0,0 +1,26 @@
+        <field name="arch" type="xml">
+            <field name='code' position="after">
+                <filter string="Uruguayan Documents" name="uy_localization" domain="[('country_id.code', '=', 'UY')]"/>
+            </field>
```
Yes we can. Here we only have the same logic we get from AR 
https://github.com/odoo/odoo/blob/7d678cc7f4c9e00d78d4120f703e90bd282c1a51/addons/l10n_ar/views/l10n_latam_document_type_view.xml#L27-L40

Do you like we remove it from the two locs? @ren-odoo

ren-odoo replied:
@zaoral we already have a default group by localization, I think that's enough.  
![image](https://github.com/odoo/odoo/assets/33298180/efd840b5-f89e-46fc-af77-75b8f63d0b2e)
Anyways users can always create a filter and save it if needed.

zaoral replied:
Hi @jco-odoo  @ren-odoo I have removed the filter.
I will mark this as resolve.

@ren-odoo Also we have a filter and a menu similar on AR Loc right now
please let me know if you want me to delete it as well in another pr 
This way we will have only one menu and one generic filter

Best Regards

ren-odoo commented on addons/l10n_uy/views/account_journal_view.xml:
```diff
@@ -0,0 +1,17 @@
+        <field name="arch" type="xml">
+            <field name="refund_sequence" position="attributes">
+                <attribute name="invisible">
+                    type not in ['sale', 'purchase'] or (l10n_latam_use_documents == True and country_code == 'UY')
```
@zaoral it's the way it works in all the latam localizations, actually it seems that it's already working like that by default so we can remove it. 

cc. @jco-odoo @fvz-odoo

jco-odoo replied:
please remove

zaoral commented on addons/l10n_uy/views/l10n_latam_document_type_views.xml:
```diff
@@ -0,0 +1,26 @@
+    <record model="ir.actions.act_window" id="action_document_type_uruguay">
+        <field name="name">Document Types</field>
+        <field name="res_model">l10n_latam.document.type</field>
+        <field name="context">{'search_default_uy_localization': 1}</field>
```
menu deleted

jco-odoo commented on addons/l10n_uy/models/account_move.py:
```diff
@@ -0,0 +1,76 @@
+        if self.journal_id.company_id.account_fiscal_country_id.code == "UY":
+            codes = self.journal_id._l10n_uy_get_journal_codes()
+            if codes:
+                domain.extend([('code', 'in', codes), ('active', '=', True)])
```
```suggestion
                domain.extend([('code', 'in', codes)])
```
That should not be necessary

jco-odoo commented on odoo/addons/base/data/res_currency_data.xml:
```diff
@@ -1689,8 +1689,8 @@
             <field name="full_name">Uruguay Peso en Unidades Indexadas</field>
-            <field name="symbol">$</field>
-            <field name="rounding">1</field>
+            <field name="symbol">UYI</field>
```
symbol is UYI?

zaoral replied:
Not really the symbol but help us to identify the currency when is been used from UYU and USD.
I will leave the suymbol as it it and we can disccuss it later

zaoral commented:
Hi @jco-odoo 
Thank for the last review.
I am working on the lasted changes.
I will making push shortly

Best Regards

zaoral commented on addons/l10n_uy/demo/res_currency_rate_data.xml:
```diff
@@ -0,0 +1,10 @@
+<?xml version='1.0' encoding='utf-8'?>
+<odoo noupdate="1">
+
+    <record id="rateUYI" model="res.currency.rate">
```
Donee

zaoral commented on addons/l10n_uy/demo/account_demo.py:
```diff
@@ -0,0 +1,20 @@
+    def _get_demo_data(self, company=False):
+        demo_data = super()._get_demo_data(company)
+        if company == self.env.ref('l10n_uy.demo_company_uy', raise_if_not_found=False):
+            # Do not load generic demo data on these companies
```
Ok, We will work on this later. We leave it as it is. and we change it later.
Best Regards

ren-odoo commented on addons/l10n_uy/data/template/account.fiscal.position-uy.csv:
Hi @zaoral @marianaluztorres can you take a looka the local fiscal positions, it seems the "Tax to Apply" is not mapped correctly. 

![image](https://github.com/odoo/odoo/assets/33298180/68f652f8-2bd7-4def-88fe-e5bd86d0bda9)

cc. @vbe-odoo @fvz-odoo

ren-odoo replied:
@zaoral  the Exportation Fiscal position also needs a change, because as it is right now, the tax on the products is removed, instead being mapped/replaced by the 0% tax. 
Can you take a look pls?
![image](https://github.com/odoo/odoo/assets/33298180/a2c54d05-a1fa-487e-afa9-860fbde5eb0b)

cc. @marianaluztorres @vbe-odoo @fvz-odoo

ren-odoo commented on addons/l10n_uy/models/l10n_latam_document_type.py:
```diff
@@ -0,0 +1,28 @@
+
+        if not serie_part or len(serie_part) > 1 or len(serie_part[0]) > 2 \
+           or not number_part or len(number_part) > 1 or len(number_part[0]) > 7:
+            raise UserError(_('Not Valid UY document number (ALPHA 2 - NUM 7)'))
```
Hi @zaoral I think this message can be improved to be more user frindly as a regular user friendly,  example:  "Please introduce a valiud Document number:   2 letters and 7 digits  (XX-0000001)"  

We can omit the UY reference as this error will be rise just in this localization.


![image](https://github.com/odoo/odoo/assets/33298180/bf58accc-83dc-45c1-84b8-12d7f15930e9)

ren-odoo commented on addons/l10n_uy/models/account_move.py:
```diff
@@ -0,0 +1,79 @@
+                lambda x: x.display_type not in ('line_section', 'line_note')):
+            vat_taxes = line.tax_ids.filtered(lambda x: x.l10n_uy_tax_category == 'vat')
+            if len(vat_taxes) != 1:
+                raise ValidationError(_('Should be one and only one VAT tax per line. Verify lines with product "%s" (Id Invoice: %s)',
```
@zaoral we can remove this validation on this module, as of now is now without EDI it shouldn't be a blocking for the user an we are also reducing the risk of missing a VAT tax now that we have fiscal positions.

We can add this an other needed validations on the EDI module. 

cc. @vbe-odoo @fvz-odoo

ren-odoo commented on addons/l10n_uy/models/account_tax.py:
```diff
@@ -0,0 +1,11 @@
+    _inherit = "account.tax"
+
+    l10n_uy_tax_category = fields.Selection([
+        ('vat', 'VAT'),
```
@jco-odoo for now we just added VAT as per the other taxes and witholds we need to go deeper in the analysis to determined what other options will be required in the selection once that we add those records in the data. 

cc. @zaoral @fvz-odoo

ren-odoo replied:
@MarianaLuzTorres @zaoral one last minor comment here, I suggest to add a tooltip, so it's clearler for the user what's the poporpuse of this field, something like this "UY: Use to group the transactions in the Financial Reports required by DGI" 

What do you think?   
cc. @fvz-odoo

ren-odoo commented on addons/l10n_uy/data/template/account.account-uy.csv:
@zaoral we're missing the translations of the default accounts:
![image](https://github.com/odoo/odoo/assets/33298180/7412a4ed-1259-4870-9fc1-adcaabb5e63e)
Can we add the value in the name@es column pls?

cc. @MarianaLuzTorres @fvz-odoo

zaoral replied:
Hi @fvz-odoo
after checking this we found out that this translation should be made by Transifex.
Not need to made changes to this PR code so I will markeit as resolved

ren-odoo commented:
Hi @jco-odoo 

All the comments on you review have been addressed by @zaoral I also did the FUN testing and besides the minor couple of comments, all the rest is ready for your final check. 

As reference here is the Testing file all the points I checked on my Test https://docs.google.com/spreadsheets/d/1gUhdN-Znr8G8aILHGjp6DMxXHj9CtVAFI6m2w053OJo/edit#gid=1700855336.

ren-odoo commented on addons/l10n_uy/data/template/account.fiscal.position-uy.csv:
Checking with @MarianaLuzTorres and @zaoral if we rather leave the Local fiscal position empty, so the default configured products on Product/product category/general settings are taken.  (as in the fiscal position we have in PE)

What do you think @jco-odoo ?

cc. @fvz-odoo

jco-odoo replied:
Yes, it does not make any sense to map them to 22% as 10% could be used locally.  Indeed, just empty should do.  (and auto_apply country UY)

jco-odoo commented on addons/l10n_uy/data/template/account.tax-uy.csv:
```diff
@@ -1,25 +1,25 @@
-"","","","","","","","","base","refund","-Tax Base Purchases","","",""
-"","","","","","","","","tax","refund","","","",""
+id,name,description,invoice_label,l10n_uy_tax_category,amount,amount_type,type_tax_use,tax_group_id,repartition_line_ids/repartition_type,repartition_line_ids/document_type,repartition_line_ids/tag_ids,repartition_line_ids/account_id,description@es,invoice_label@es
+vat1,22%,VAT Sales (22%),VAT Sales (22%),vat,22,percent,sale,tax_group_iva_22,base,invoice,+Base Sales 22%,,IVA Ventas (22%),IVA Ventas (22%)
```
please export with ""

jco-odoo commented:
Please check https://runbot.odoo.com/runbot/build/52047885 as well

fvz-odoo commented:
Hello @zaoral 

Thanks for all the changes, I tested and LGTM. We just have pending the thing we discussed about the translation. 

@jco-odoo this is ready for your final review. We can talk about the issue of the translation of some default accounts in our meeting. 

Let us know if anything please!

Fernando

@ren-odoo @pablohmontenegro

jco-odoo commented:
@zaoral Ok, I think we just need a rebase to get everything green.  Also, the commit message [ADD] l10n_uy could be improved as we do not really add the module.  I think @fvz-odoo put like a list of things done in the messages above.

zaoral commented:
HI @jco-odoo 
Thank you for your message.
I have rebased the PR and also changed the commit message.
Please review and let me know if it is ok.

Best Regards

jco-odoo commented:
@robodoo r+ rebase-ff
